projects/igniteui-angular/src/lib/buttonGroup/buttonGroup.component.ts

```diff
@@ -462,14 +462,19 @@ export class IgxButtonGroupComponent extends DisplayDensityBase implements After
             });
         };
 
-        this.viewButtons.changes.pipe(takeUntil(this.queryListNotifier$)).subscribe(() => initButtons());
-        this.templateButtons.changes.pipe(takeUntil(this.queryListNotifier$)).subscribe(() => initButtons());
-        initButtons();
-
-        this._cdr.detectChanges();
-
         this.mutationObserver = this.setMutationsObserver();
 
+        this.viewButtons.changes.pipe(takeUntil(this.queryListNotifier$)).subscribe(() => {
+            this.mutationObserver.disconnect();
+            initButtons();
+        });
+        this.templateButtons.changes.pipe(takeUntil(this.queryListNotifier$)).subscribe(() => {
+            this.mutationObserver.disconnect();
+            initButtons();
```

Comment:
After InitButtons you have to re-attach the mutation observer (like on line 478)